src/metabase/api/collection.clj

```diff
@@ -45,17 +45,68 @@
 
 (declare root-collection)
 
+(defn- remove-other-users-personal-collections
+  "Remove from the provided collections any that are personal and belong to another user.
+   Will work in a single pass if collections are already ordered such that root (location = /)
+   collections appear before children, but will catch any mis-ordered entries and
+   do a final pass on those if needed."
+  ([collections]
+   (remove-other-users-personal-collections (:id @api/*current-user*) collections))
```

Comment:
in the future you can use `metabase.api.common/*current-user-id*` or feel free to change it now if you like.

Reply:
Done